xgcm/grid.py

```diff
@@ -1624,6 +1624,9 @@ def pad(self, *arrays, boundary_width=None, boundary=None, fill_value=None):
                 if ax_boundary == "extrapolate":
                     # TODO implement extrapolation
                     raise NotImplementedError
+                elif ax_boundary is None:
+                    # TODO this is necessary, but also seems inconsistent with the docstring, which says that None = "no boundary condition"
+                    ax_boundary = "periodic"
```

Comment:
Why is this necessary? I think the problem is that in the init of the Grid class `periodic` is usually True, and a lot of tests will thus implicitly assume a periodic boundary. 
The simplest fix to that might be to define an override [somewhere here](https://github.com/xgcm/xgcm/blob/95f4f33d72d2add00136e27f6b3bedecb97d4d77/xgcm/grid.py#L1147-L1158) that basically does something like: 
```
# for each axis:
if isperiodic:
    boundary = 'periodic'
```
That way we internally drop the seperate kwarg `periodic` but the user can (for now) still use the existing API.

That should hopefully make all legacy tests pass. I am curious if after that change this is still necessary. I would like to avoid setting periodic here, and rather spit out an error if boundary is still none (after the changes suggested above).

Reply:
> Why is this necessary? I think the problem is that in the init of the Grid class periodic is usually True, and a lot of tests will thus implicitly assume a periodic boundary.

I think its something like that - related to the redundancy of the `periodic` argument that you mentioned last week.

> That way we internally drop the seperate kwarg periodic but the user can (for now) still use the existing API.

This seems like a good idea.

> I would like to avoid setting periodic here, and rather spit out an error if boundary is still none (after the changes suggested above).

Using `boundary=None` to mean a specific boundary condition choice is sort of inconvenient because to be able to override the Axis' default choice of bc you want the default value of the boundary kwarg in `diff` etc. to be `None` as in "no value provided".

Reply:
> Why is this necessary?

It's necessary because if you don't add something like this the existing periodic tests raise an error from `.pad` that `boundary=None` is not a supported type of boundary.

Reply:
> It's necessary because if you don't add something like this the existing periodic tests raise an error from .pad that boundary=None is not a supported type of boundary.

But they hopefully would not do that anymore if the default `periodic` is propagated from the init call of `Grid`?